src/storage/StorageModule.ts

```diff
@@ -1,32 +1,319 @@
+    private blockConfirmationStructsMap: Map<string, BlockConfirmationStruct>;
+    private latestBlockConfirmationKey: string;
+
+    // map [fork count, block height] => struct
+    // we make the key a string to avoid double mapping
+    private stateSnapshotStructsMap: Map<string, StateSnapshotStruct>;
+    //map [stateSnapshotHash] => [fork count, block height]
+    private hashToStateSnapshotKeyMap: Map<BytesLike, string>;
+    //could be replaced with a key, but we would need some computation
+    private latestOnChainStateSnapshot: {
+        stateSnapshot: StateSnapshotStruct;
+        timestamp: number;
+    };
+
+    private joinChannelBlockHashesMap: Map<string, BytesLike>;
+    //TODO: check if BytesLike is more efficient than string
+    // we can either store the hash or the key to the joinChannelBlockHashesMap data
+    private latestJoinChannelBlockHash: BytesLike;
+
+    //map [blockHash] => ExitChannelBlockStruct
+    private exitChannelBlockMap: Map<string, ExitChannelBlockStruct>;
+    //TODO: check if BytesLike is more efficient than string
+    // we can either store the hash or the key to the exitChannelBlockHashesMap data
+    private latestExitChannelBlockHash: BytesLike;
+
+    private totalDeposits: BalanceStruct;
+    private totalWithdrawals: BalanceStruct;
+
+    constructor() {
+        //TODO: This should be replaced by the actual genesis state snapshot
+        this.latestOnChainStateSnapshot = {
+            stateSnapshot: {
+                stateMachineStateHash: "0x",
+                participants: [],
+                forkCnt: 0,
+                latestJoinChannelBlockHash: "0x",
+                latestExitChannelBlockHash: "0x",
+                totalDeposits: {
+                    amount: BigInt(0),
+                    data: "0x"
+                },
+                totalWithdrawals: {
+                    amount: BigInt(0),
+                    data: "0x"
+                }
+            },
+            timestamp: 0
+        };
+        this.blockConfirmationStructsMap = new Map();
+        this.stateSnapshotStructsMap = new Map();
+        this.joinChannelBlockHashesMap = new Map();
+        this.exitChannelBlockMap = new Map();
+        this.totalDeposits = {
+            amount: BigInt(0),
+            data: "0x"
+        };
+        this.totalWithdrawals = {
+            amount: BigInt(0),
+            data: "0x"
+        };
+        this.latestJoinChannelBlockHash = ethers.ZeroHash;
+        this.latestExitChannelBlockHash = ethers.ZeroHash;
+        this.hashToStateSnapshotKeyMap = new Map();
+        this.latestBlockConfirmationKey = "0";
+        console.log("StateSnapshotStorage initialized");
+    }
+
+    //#region Block confirmation management methods
+    getPreviousBlockHash(
+        forkCnt: number,
+        transactionCnt: number
+    ): string | undefined {
         // TODO
         return ethers.ZeroHash;
     }
 
+    getLatestBlock(): UnrolledSignedBlock {
```

Comment:
I thought we're not using these artificial data structures 'Unrolled' and similar

Reply:
> I thought we're not using these artificial data structures 'Unrolled' and similar

yeah i used what was already done in the dispute branch, but i can easily use the usual datatypes instead

Reply:
Yes please do

Reply:
We can have helper functions for common operations that need to return multiple things (perform multiple reads) - the StorageManager can implement them, but first let's finish the storage layout (and partition) so it's clear what is persisted and how to CRUD (create, read, update, delete)

Reply:
yeah now that the refactoring is done these helper functions will be easier to make on the top level storageManager

Reply:
@daiagi how do you want the data ? giving you a tuple ? Since creating an unrolled typed is nah

Reply:
![Screenshot from 2025-06-05 11-04-37](https://github.com/user-attachments/assets/89e86acb-dd2e-44b4-b0e7-8d2573dfa62d)

Reply:
this ok ?

Reply:
yep. this will work